agent/hcp/telemetry/otel_sink.go

```diff
@@ -23,12 +22,15 @@ import (
 // DefaultExportInterval is a default time interval between export of aggregated metrics.
 const DefaultExportInterval = 10 * time.Second
 
+type ConfigProvider interface {
+	GetLabels() map[string]string
+	GetFilters() *regexp.Regexp
+}
+
 // OTELSinkOpts is used to provide configuration when initializing an OTELSink using NewOTELSink.
 type OTELSinkOpts struct {
-	Reader  otelsdk.Reader
-	Ctx     context.Context
```

Comment:
After discussion with @chapmanc , preferred to not add ctx in struct so also removed that. It was just used for a logger in the OTELSink